polynote-kernel/src/main/scala/polynote/kernel/interpreter/scal/ScalaCompleter.scala

```diff
@@ -222,26 +185,125 @@ class ScalaCompleter[Compiler <: ScalaCompiler](
   private def isVisibleSymbol(sym: Symbol) =
     sym.isPublic && !sym.isSynthetic && !sym.isConstructor && !sym.isOmittablePrefix && !sym.name.decodedName.containsChar('$')
 
-  private def deepestSubtreeEndingAt(topTree: Tree, offset: Int): Tree = {
-    var deepest: Tree = topTree
-    val traverser: Traverser = new Traverser {
-      private var depth = 0
-      private var deepestDepth = -1
-      override def traverse(tree: compiler.global.Tree): Unit = {
-        if (tree.pos != null && tree.pos.isDefined && !tree.pos.isTransparent && (tree.pos.source eq topTree.pos.source) && tree.pos.end == offset && depth >= deepestDepth) {
-          deepest = tree
-          deepestDepth = depth
+  private def deepestSubtreeEndingAt(stats: List[Tree], pos: Position): Option[Tree] = {
```

Comment:
This is simplified because it's delegated to the PC, but is needlessly complicated syntactically just for debugability purposes.